models/balances/ethereum/erc721/balances_ethereum_erc721_hour.sql

```diff
@@ -17,23 +17,41 @@ with
             ) as hour
     )
 
-, daily_balances as
+, hourly_balances as
  (SELECT
-    wallet_address,
-    token_address,
-    tokenId,
-    hour,
-    lead(hour, 1, now()) OVER (PARTITION BY token_address, tokenId ORDER BY hour) AS next_hour
-    FROM {{ ref('transfers_ethereum_erc721_rolling_hour') }})
+    b.wallet_address,
+    b.token_address,
+    b.tokenId,
+    b.hour,
+    b.amount, 
+    lead(b.hour, 1, now()) OVER (PARTITION BY b.wallet_address, b.token_address, b.tokenId ORDER BY hour) AS next_hour
+FROM {{ ref('transfers_ethereum_erc721_rolling_hour') }} b
+LEFT JOIN {{ ref('balances_ethereum_erc721_noncompliant') }}  as nc
+    ON b.token_address = nc.token_address
+WHERE nc.token_address IS NULL 
+)
 
-SELECT distinct
+SELECT 
+    blockchain, 
+    hour,
+    wallet_address, 
+    token_address,
+    tokenId, 
+    collection
+FROM 
+(
+SELECT 
     'ethereum' as blockchain,
     d.hour,
     b.wallet_address,
     b.token_address,
     b.tokenId,
+    SUM(amount) as amount, 
```

Comment:
I'm wondering if we need a sum() here, I think we could just get by with defining
``where amount = 1``

Reply:
hmm true lemme look into that. There was originally a **Where amount = 1** filter in the hourly/daily balance table (where we did the lead window function). 

Somehow it affected the results of the query causing wallets that don't own a tokenid to get counted multiple times 😅. Maybe the amount filter should just be below. 

I tried mirroring getting balances for an erc20 token (hence the SUM(amount)).